onnxruntime/test/perftest/ort_test_session.cc

```diff
@@ -136,5 +136,68 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
   }
 }
 
+bool OnnxRuntimeTestSession::PopulateGeneratedInputTestData()
+{
+  size_t num_input_nodes;
+  num_input_nodes = session_.GetInputCount();
+
+  // iterate over all input nodes
+  for (size_t i = 0; i < num_input_nodes; i++) {
+    Ort::TypeInfo typeInfo = session_.GetInputTypeInfo(i);
+    Ort::MemoryInfo memory_info = Ort::MemoryInfo::CreateCpu(OrtArenaAllocator, OrtMemTypeDefault);
+    if (typeInfo.GetONNXType() == ONNX_TYPE_TENSOR) {
+        auto tensorInfo = typeInfo.GetTensorTypeAndShapeInfo();
+        std::vector<int64_t> input_node_dim = tensorInfo.GetShape();
```

Comment:
How does this handle free dimensions?  I assume the intention is to convert them to 1s like Winmlrunner?

Reply:
Oh right, this current implementation doesn't handle free dimensions. Let me look into it more

Reply:
I will treat free dimensions as 1.